lib/logstash/outputs/elasticsearch/template_manager.rb

```diff
@@ -15,16 +15,24 @@ def self.install_template(plugin)
                            "The legacy template API is slated for removal in Elasticsearch 9.")
       end
 
+      # we should avoid installing defaul legacy templates for ES8+ versions
+      # we need to guide customers using custom template or composable template API
+      if !plugin.template && plugin.template_api == 'legacy' && plugin.maximum_seen_major_version >= 8
+        raise LogStash::ConfigurationError, "Plugin cannot automatically resolve the default template when using legacy template API. Please consider using custom template or composable template API."
+      end
+
       if plugin.template
         plugin.logger.info("Using mapping template from", :path => plugin.template)
+        # we let customers to place ILM setting in their custom templates
         template = read_template_file(plugin.template)
       else
         plugin.logger.info("Using a default mapping template", :es_version => plugin.maximum_seen_major_version,
                                                                :ecs_compatibility => plugin.ecs_compatibility)
         template = load_default_template(plugin.maximum_seen_major_version, plugin.ecs_compatibility)
+        # we set ILM setting when we automatically resolve template
+        add_ilm_settings_to_template(plugin, template) if plugin.ilm_in_use?
```

Comment:
We have [another logic to set ILM (elasticsearch.rb#L350)](https://github.com/mashhurs/logstash-output-elasticsearch/blob/0eff5a8b2bce2cf3f585a9fb508219b252b5be5c/lib/logstash/outputs/elasticsearch.rb#L350) and I wonder if we need to change this behavior as well. It seems to me we need to ignore ILM set process if custom template is provided.
```
setup_ilm if ilm_in_use? && !@template
```